app/models/concerns/declarative_metadata.rb

```diff
@@ -0,0 +1,15 @@
+module DeclarativeMetadata
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def refresh_metadata(target, when_changing: [], on_event: %i[update])
+      after_commit(on: on_event) do
+        should_touch = when_changing.blank? || when_changing.any? do |attr|
+          saved_change_to_attribute?(attr)
+        end
+
+        Metadata::Manager.new.refresh_metadata!(instance_exec(&target)) if should_touch
+      end
+    end
```

Comment:
Based on: https://github.com/DFE-Digital/register-early-career-teachers-public/pull/1260/files#diff-07f753084c94c0d0d5823e7662a1400fd534a80bcac415fb6a1c0c60c3573fdaR30

`instance_exec(&target) ` could potentially return nil, i think we will need to check for that here before running `refresh_metadata!`